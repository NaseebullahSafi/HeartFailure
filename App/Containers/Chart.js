import React, {Component} from 'react'

// predefined component of react
import {Dimensions} from 'react-native'

// strict types for props
import PropTypes from 'prop-types'

// we need the highcharts support in react native
import ChartView from 'react-native-highcharts'

// this is important, the name should be specified as below
const Highcharts = "Highcharts"

/**
 * This component will take the config object from the props, and
 * pass it to highcharts to visualise. The object must of the
 * same structure as highcharts predefined obj
 * ==============================================================
 */
export default class Chart extends Component {
  constructor(props) {
    super(props);

    /**
     * Let us identify unique props for this component
     * @type {{
     *  type        : string, height  : Number, config: Object,
     *  chartLoading: boolean, options: {global: {useUTC: boolean}}
     * }}
     */
    this.state = {
      // the default type for the graph is day
      type: "day",

      // the chart needs to have specific height
      height: this.props.height,

      // this is the object that comes from other components
      config: this.props.config,

      // important for checking if the graph has loaded
      chartLoading: true,

      // highcharts default params for options
      options: {
        global: {
          useUTC: false
        }
      }
    }
  };

  /**
   * When the component loads, tell others that the component is
   * ready
   * ==============================================================
   */
  componentDidMount() {
    this.setState({loading: true});
  }

  /**
   * Default params for highcharts
   * ==============================================================
   */
  options = () => {
    this.setState({
      options: {
        global: {
          useUTC: false
        }
      }
    })
  };


  /**
   * When the component is ready, pass it to highcharts for
   * visualisation
   * ==============================================================
   * @return {XML}
   */
  render() {
    return(
      <ChartView
        style={{
          height: this.props.height,
          width: this.props.width || Dimensions.get('window').width
        }}
        config={this.props.config}
        options={this.props.options}
        // important if we want the chart to have scrollbar
        stock={this.props.stock || false}
        scalesPageToFit={false}
        automaticallyAdjustContentInsets={false}
        showsHorizontalScrollIndicator={false}
        horizontal={false}
        javaScriptEnabled={true}
        domStorageEnabled={true}
      />
    )
  }
}

// let us define some types for our props
Chart.propTypes = {
  height        : PropTypes.number.isRequired,
  config        : PropTypes.object.isRequired,
};
