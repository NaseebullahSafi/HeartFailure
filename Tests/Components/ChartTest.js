// need to import React itself
import React from 'react'

// everything fro react native
import 'react-native'

// required for mocking modules
import Enzyme,{ shallow } from 'enzyme'

// needed to pass components to the renderer
import renderer from 'react-test-renderer'

// adapter is needed for this react version
import Adapter from 'enzyme-adapter-react-16';

// the component that will be tested
import Chart from "../../App/Containers/Chart";

Enzyme.configure({ adapter: new Adapter() });

describe('Testing <Chart /> component', () => {
  // create an image of the component, and match it with the snapshot
  it('renders as expected', () => {
    const tree = renderer.create(
      <Chart height={100} config={{}}/>
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });
});

