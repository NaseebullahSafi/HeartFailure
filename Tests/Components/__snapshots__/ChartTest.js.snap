// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Chart /> component renders as expected 1`] = `
<View
  style={
    Object {
      "height": 100,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={undefined}
      automaticallyAdjustContentInsets={false}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={undefined}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "baseUrl": "web/",
          "html": "<html>
                    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=0\\" />
                    <style media=\\"screen\\" type=\\"text/css\\">
                    #container {
                        width:100%;
                        height:100%;
                        top:0;
                        left:0;
                        right:0;
                        bottom:0;
                        position:absolute;
                        user-select: none;
                        -webkit-user-select: none;
                    }
                    </style>
                    <head>
                        <script src=\\"https://code.jquery.com/jquery-2.1.4.min.js\\"></script>
                        <script src=\\"https://code.highcharts.com/highcharts.js\\"></script>
                        
                        
                        <script src=\\"https://code.highcharts.com/modules/exporting.js\\"></script>
                        <script>
                        $(function () {
                            Highcharts.setOptions(undefined);
                            Highcharts.chart('container', }           );
                        });
                        </script>
                    </head>
                    <body>
                        <div id=\\"container\\">
                        </div>
                    </body>
                </html>",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "height": 100,
            "width": 750,
          },
        ]
      }
    />
  </View>
</View>
`;
